### ISSUE TITLE
v4.3+ hangs on otter assign when using "converted" notebooks

### ISSUE THREAD

**Issue by sean-morris** [ucbds-infra/otter-grader#635]
**Describe the bug**
I have notebooks that have been converted via the 'otter.assign.v0.convert' tool. When using version 4.3+ the `otter assign` command hangs on notebooks that have been converted.

It appears to hang on the step between Generating Views and Generating autograder zipfile. I would assume it is stuck on the zip step but it is hard to tell. 

In both cases, one in which we use 4.3.2 and one in which we use 4.2.0, we do get the nbconvert/nbformat warning: `MissingIDFieldWarning: Code cell is missing an id field, this will become a hard error in future nbformat versions. You may want to use `normalize()` on your notebooks before validations (available since nbformat 5.1.4). Previous versions of nbformat are fixing this issue transparently, and will stop doing so in the future.
  validate(nb)`

Not sure if this is relevant. 

**To Reproduce**
1. Convert a notebook from v0 to v1
2. Install the lastest otter-grader
3. Run otter assign
4. Process Hangs.

**Expected behavior**
otter assign should complete or give an error as to what might be the problem

**Versions**
otter 4.3.2 or 4.3.1
python 3.10.8

**Additional context**
I have reproduced this problem on two different sets of notebooks. The problem for me with going back to 4.2.0 is that I need the  run-on: colab and tests: files assignment configs.

**Comment by sean-morris** [comment]
@chrispyles 

This is causing a series of problems. We can't convert Data 8 notebooks and then run otter assign with 4.3+ which then means we use 4.2 but as a result have to manually update the autograder.zip requirements.txt file by changing sklearn to scikit-learn manually. We can do it. 

But do have any sense as to why it is hanging? Point me in a direction and I can take a look.

**Comment by sean-morris** [comment]
@taylorgibson ok yes #634 looks like the problem. I reverted to 4.2 as well but that causes other issues! I am going to close this in deference to a much more detailed understanding of this problem at #634
